base/server/python/pki/server/deployment/pkihelper.py

```diff
@@ -2344,14 +2344,17 @@ def deregister(self, critical_failure=False):
     def execute_using_pki(
             self, tkshost, tksport, subsystemnick,
             token_pwd, tpshost, tpsport, critical_failure=False):
-        command = ["/usr/bin/pki",
-                   "-p", str(tksport),
-                   "-h", tkshost,
+
+        tks_url = 'https://%s:%s' % (tkshost, tksport)
+        password_conf = os.path.join(
+            self.mdict['pki_instance_configuration_path'],
+            'password.conf')
+
+        command = ["pki",
+                   "-U", tks_url,
                    "-n", subsystemnick,
-                   "-P", "https",
                    "-d", self.mdict['pki_server_database_path'],
-                   "-c", token_pwd,
-                   "-t", "tks",
```

Comment:
Just wondering -- `tks_url` is created out of `tkshost` and `tksport` but no subsystem path. We're removing the `tks` subsystem type here as well. Does `tks-tpsconnector` infer that its a `tks` subsystem type now? Why remove this line?

Reply:
Yes, the `-U <URL>` only needs the hostname & port. The subsystem path is determined by the command we're executing. This way we can use the same URL for all commands against the same server.